src/public/windowObject.ts

```diff
@@ -0,0 +1,60 @@
+import { ensureInitialized, sendMessageRequest } from "../internal/internalAPIs";
+import { GlobalVars } from "../internal/globalVars";
+import { frameContexts } from "../internal/constants";
+
+export interface IWindowObject {
+  postMessage(message): void;
+  addEventListener(type: string, listener: Function): void;
+}
+
+export class ChildWindowObject implements IWindowObject {
+  public postMessage(
+    message: any
+  ): void {
+    ensureInitialized();
+    const messageId = sendMessageRequest(GlobalVars.parentWindow, "messageForChild", [
+      message
+    ]);
+
+    GlobalVars.callbacks[messageId] = (success: boolean, result: string) => {
+      if (!success) {
+        throw new Error(result);
+      }
+    };
+  }
+
+  public addEventListener(type, listener) {
```

Comment:
NIT: Add types for `type` and `listener` as well as return type for `addEventListener` method